pkg/controller/builder.go

```diff
@@ -253,23 +254,61 @@ func UpdateDeploymentProxyHook(pc clients.ProxyClientInterface) deploymentcontro
 	}
 }
 
-func setContainerArg(con *corev1.Container, arg, value string) {
+// The return behavior:
+// 1. If the input arg value matches the container argument = success
+// 2. If the input arg value does not match the container argument = error
+func setContainerArg(con *corev1.Container, arg, value string) error {
 	// Need to remove any existing `arg` arguments first
 	// This could happen because the experimental manifest may have feature gates already enabled
 	var args []string
+	var containerValues []string
 	log := klog.FromContext(context.Background()).WithName("builder")
 
+	// Calculate the container argument value and create a new container argument list without it
 	for _, a := range con.Args {
 		if !strings.HasPrefix(a, arg) {
 			args = append(args, a)
 		} else {
-			log.Info("Removing arg", "container", con.Name, "argument", a)
+			// Save off original argument value by removing prefix
+			s := strings.TrimPrefix(a, arg)
+			s = strings.TrimPrefix(s, "=")
+			log.Info("Removing argument", "container", con.Name, "arg", arg, "value", s)
+			if s != "" {
+				containerValues = append(containerValues, strings.Split(s, ",")...)
+			}
 		}
 	}
 
+	containerValue := ""
+	if len(containerValues) > 0 {
+		sort.Strings(containerValues)
+		containerValue = strings.Join(containerValues, ",")
+	} else {
+		// Otherweise, there were no original arguments, so just set what was passed in
+		log.Info("No existing arguments of type", "arg", arg)
+	}
+
+	// Check to see if the container values match the new values
+	if value != containerValue {
+		// Arguments do not match, so return an error.
+		// The likely reason for this error is that the feature flag coding has not completed yet;
+		// either the cluster-olm-operator mapping is not present,
+		// or the feature was not added to the experimental manifest.
+		err := fmt.Errorf("new and existing arguments do not match")
+		log.Error(err, "New and existing arguments do not match", "arg", arg, "new", value, "existing", containerValue)
```

Comment:
This log line will look something like this:
```
"ERROR: New and existing arguments do not match", "err": "new and existing arguments do not match", "arg": "foo", "new": "bar1", "existing": "bar2"
```

Note the duplicate error text. Do we need this log line (or any log line in this function)? Now that there is no warning possibility, it seems like either everything is good or we return an error, that will eventually be logged somewhere.

Reply:
Ywah, I knew it would be duplicate test. I removed all the logs; they were only useful for debugging.